client-default/src/main/scala/com.gu.contentapi.client/GuardianContentClient.scala

```diff
@@ -8,12 +8,13 @@ import scala.concurrent.{ExecutionContext, Future, Promise}
 
 class GuardianContentClient(val apiKey: String) extends ContentApiClient {
 
-  private val http = new OkHttpClient.Builder()
+  protected def httpClientBuilder = new OkHttpClient.Builder()
```

Comment:
I'm on the fence here. Do we expose `OkHttp` to the outside world or provide a layer of abstraction by means of a factory method? I suppose, as the intent of this default client is precisely to provide a simple implementation, we can opt for the former, but in another instance I'd prefer the latter.

Reply:
I think I'd still prefer to see a pattern where the constructor and `followRedirects` calls are hidden away.

Reply:
Sorry, Regis, I'm not sure what you mean by hiding the constructor and `followRedirects` away...

Reply:
Intuitively, I understand you want to customise the configuration of the builder for your needs, which is sensible. I was wondering if something like this would be a better approach:
```scala
def withOptions: OkHttpClient.Builder => OkHttpClient.Builder =
  _.connectTimeout(1, TimeUnit.SECONDS)
    .readTimeout(2, TimeUnit.SECONDS)
    .connectionPool(new ConnectionPool(10, 60, TimeUnit.SECONDS))
private val httpClientBuilder = withOptions(new OkHttpClient.Builder()).followRedirects(true)
```

Reply:
I see how this would work but I'm not sure what value it adds...

Reply:
In a future where we might want to enforce specific HTTP/S settings, giving access to the underlying implementation would allow anyone to discard those settings.

Reply:
We'd have to enforce these settings on the server, not the client. The client is open-source _and_ pluggable so a developer can always copy the class and customise it.

Additionally, any required settings should be declared in the docs of `ContentApiClient#get`.